cluster/manager.go

```diff
@@ -184,20 +188,29 @@ func (c *MemberlistManager) clusterStats() {
 	primNotReady := 0
 	secReady := 0
 	secNotReady := 0
+	queryReady := 0
+	queryNotReady := 0
 	partitions := make(map[int32]int)
 	for _, p := range c.members {
+		parts := p.GetPartitions()
 		if p.Primary {
 			if p.IsReady() {
 				primReady++
 			} else {
 				primNotReady++
 			}
-		} else {
+		} else if len(parts) > 0 {
```

Comment:
instead of just assuming that nodes without partitions are query nodes, wouldn't it be nicer to change `cluster.HTTPNode.Primary` into some property that can represent if a member is `primary/secondary/query`? Or would that make the upgrade procedure from the old MT version to this one too complicated?

Reply:
If the problem is the upgrade procedure, then I'd at least abstract that assumption that a member without partitions is a query node. F.e. `HTTPNode` could have a method like `IsQuery()` or something like that which returns `true` if `len(partitions) == 0`, this would make this code easier to understand.

Reply:
> wouldn't it be nicer to change cluster.HTTPNode.Primary into some property that can represent if a member is primary/secondary/query

yeah, or maybe not the .Primary attribute, but another attribute to mark the cluster mode (full/shard/query)

> I'd at least abstract that assumption that a member without partitions is a query node. F.e. HTTPNode could have a method like IsQuery()

Good idea. I can give this a shot

Reply:
hmm I think my approach would still not be as robust as it should.
you can set your kafka-mdm-in partitions to "*", which causes it to query kafka for available partitions.
If kafka says that there are no partitions, i wouldn't want the instance to be treated as if it's a query node.
so i think now we should make this more robust by being explicit and tracking the cluster mode in the gossiped node information.

Reply:
yeah, that sounds like a better idea

Reply:
ok, got this done now. PTAL

Reply:
Yes, that's much better